src/libsyntax/parse/mod.rs

```diff
@@ -306,128 +311,49 @@ pub fn stream_to_parser(sess: &ParseSess, stream: TokenStream) -> Parser<'_> {
     Parser::new(sess, stream, None, true, false)
 }
 
+fn shrink(span: Span, d_lo: u32, d_hi: u32) -> Span {
+    span.with_lo(span.lo() + BytePos(d_lo)).with_hi(span.hi() - BytePos(d_hi))
+}
+
 /// Parses a string representing a character literal into its final form.
 /// Rather than just accepting/rejecting a given literal, unescapes it as
 /// well. Can take any slice prefixed by a character escape. Returns the
 /// character and the number of characters consumed.
-fn char_lit(lit: &str, diag: Option<(Span, &Handler)>) -> (char, isize) {
-    use std::char;
-
-    // Handle non-escaped chars first.
-    if lit.as_bytes()[0] != b'\\' {
-        // If the first byte isn't '\\' it might part of a multi-byte char, so
-        // get the char with chars().
-        let c = lit.chars().next().unwrap();
-        return (c, 1);
-    }
-
-    // Handle escaped chars.
```

Comment:
I have no idea why the unescaping code was duplicated.
Both versions are ancient, AFAIC.